Issue #899 fix

jonsedar opened the pull request:
Added handling for `ax` kwarg, as per issue https://github.com/pymc-devs/pymc3/issues/899

Also:
1. Added `figsize` kwarg with handling similar to `traceplot()`
2. Renamed kwarg `vars` to `varnames` because `vars` is an alias for
`locals` in iPython
3. Replaced `xrange` with `range` for Python3 compliance
4. Changed return object from tuple of `(fig,ax)` to simply `ax`
similar to `traceplot()`
5. Clarified a couple of variables to indicate their purpose: `chains`
`nchains`, `val` to `varname`.

Commit:
Fix for issue #899 autocorrplot clobbers axes

Added handling `ax` kwarg, as per issue.

Also:
1. Added `figsize` kwarg with handling similar to traceplot()
2. Renamed kwarg `vars` to `varnames` because `vars` is an alias for
`locals` in iPython
3. Replaced `xrange` with `range` for Python3 compliance
4. Changed return object from tuple of (`fig`,`ax`) to simply `ax`
similar to `traceplot()`
5. Clarified a couple of variables to indicate their purpose: `chains`
`nchains`, `val` to `varname`.

Commit:
Merge remote-tracking branch 'pymc-devs/master' into issue_899_fix

twiecki commented:
```
======================================================================
ERROR: pymc3.tests.test_plots.test_multichain_plots
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda2/envs/testenv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/pymc-devs/pymc3/pymc3/tests/test_plots.py", line 59, in test_multichain_plots
    autocorrplot(ptrace, vars=['switchpoint'])
TypeError: autocorrplot() got an unexpected keyword argument 'vars'
```

jonsedar commented:
Bugger, apologies - I'll fix that test

Commit:
Merge remote-tracking branch 'pymc-devs/master' into issue_899_fix

Commit:
edited call to autocorrplot() in test

I previously made a change to the function signature for
autocorrplot(), which broke a test. I’ve updated that test now.

Commit:
updated function calls in examples

Some examples were failing due to the changed function signature, I’ve
edited accordingly

twiecki commented:
LGTM, thanks!
